src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -187,109 +194,151 @@ public class MicrosoftIdentityPlatformApplicationManager
+            if (appUpdates != null)
+            {
+                try
                 {
-                    Web = existingApplication.Web
-                };
+                    // TODO: update other fields, see https://github.com/jmprieur/app-provisonning-tool/issues/10
+                    var updatedApp = await graphServiceClient.Applications[remoteApp.Id].Request().UpdateAsync(appUpdates);
+                    if (updatedApp is null)
+                    {
+                        return new JsonResponse(commandName, State.Fail, $"{Resources.FailedToUpdateApp} {remoteApp.DisplayName} {remoteApp.AppId}");
+                    }
 
-                //update redirect uris
-                List<string> existingRedirectUris = updatedApp.Web.RedirectUris.ToList();
-                List<string> urisToEnsure = ValidateUris(toolOptions.RedirectUris).ToList();
-                int originalUrisCount = existingRedirectUris.Count;
-                existingRedirectUris.AddRange(urisToEnsure);
-                updatedApp.Web.RedirectUris = existingRedirectUris.Distinct();
-                if (updatedApp.Web.RedirectUris.Count() > originalUrisCount)
-                {
-                    needsUpdate = true;
+                    return new JsonResponse(commandName, State.Success, $"{Resources.SuccessfullyUpdatedApp} {updatedApp.DisplayName} ({updatedApp.AppId})");
                 }
-
-                if (updatedApp.Web.ImplicitGrantSettings == null)
+                catch (ServiceException se)
                 {
-                    updatedApp.Web.ImplicitGrantSettings = new ImplicitGrantSettings();
+                    return new JsonResponse(commandName, State.Fail, se.Error?.Message);
                 }
+            }
 
-                //update implicit grant settings if need be.
-                if (toolOptions.EnableAccessToken.HasValue && (toolOptions.EnableAccessToken.Value != updatedApp.Web.ImplicitGrantSettings.EnableAccessTokenIssuance))
-                {
-                    needsUpdate = true;
-                    updatedApp.Web.ImplicitGrantSettings.EnableAccessTokenIssuance = toolOptions.EnableAccessToken.Value;
-                }
+            return new JsonResponse(commandName, State.Success, $"{Resources.NoUpdateNecessary} {remoteApp.DisplayName} {remoteApp.AppId}");
+        }
 
-                if (toolOptions.EnableIdToken.HasValue &&
-                    (toolOptions.EnableIdToken.Value != updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance))
+        /// <summary>
+        /// Updates application in Azure AD, returns updated app or null if no updates were made
+        /// </summary>
+        /// <param name="existingApplication"></param>
+        /// <param name="toolOptions"></param>
+        /// <returns></returns>
+        private Application? UpdateApplication(Application existingApplication, ProvisioningToolOptions toolOptions, bool isBlazorWasm = false)
```

Comment:
Would it be possible to update both `UpdateApplication` method's summaries with a description of the flow of changes. It's quite confusing to read through with very basic comments. Not sure why we are calling certain things.